src/core/Form/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,482 @@
+import { DatePicker } from './DatePicker/DatePicker';
+import { Label, LabelMode } from '../Label/Label';
+import { StatusText } from '../StatusText/StatusText';
+import { HintText } from '../HintText/HintText';
+import { VisuallyHidden } from '../../VisuallyHidden/VisuallyHidden';
+import { Icon } from '../../Icon/Icon';
+import { InputStatus, StatusTextCommonProps } from '../types';
+import { baseStyles } from './DateInput.baseStyles';
+import {
+  datePickerDefaultTexts,
+  DatePickerTextProps,
+  InternalDatePickerTextProps,
+} from './datePickerTexts';
+import {
+  defaultDateAdapter,
+  moveYears,
+  DateAdapter,
+  dayIsInRange,
+  firstDayOfMonth,
+  lastDayOfMonth,
+  cellDateAriaLabel,
+} from './dateUtils';
+import { getLogger } from '../../../utils/log';
+
+const baseClassName = 'fi-date-input';
+export const dateInputClassNames = {
+  baseClassName,
+  fullWidth: `${baseClassName}--full-width`,
+  hasPicker: `${baseClassName}--has-picker`,
+  disabled: `${baseClassName}--disabled`,
+  error: `${baseClassName}--error`,
+  success: `${baseClassName}--success`,
+  labelIsVisible: `${baseClassName}_label--visible`,
+  inputAndPickerWrapper: `${baseClassName}_input-and-picker-wrapper`,
+  inputElementContainer: `${baseClassName}_input-element-container`,
+  inputElement: `${baseClassName}_input`,
+  pickerElementContainer: `${baseClassName}_picker-element-container`,
+  pickerButton: `${baseClassName}_picker-button`,
+  pickerButtonDisabled: `${baseClassName}_picker-button--disabled`,
+  pickerIcon: `${baseClassName}_picker-icon`,
+  styleWrapper: `${baseClassName}_wrapper`,
+  statusTextHasContent: `${baseClassName}_statusText--has-content`,
+};
+
+export interface DatePickerProps {
+  /** Enables date picker for date input.
+   * @default false
+   */
+  datePickerEnabled?: boolean;
+  /**
+   * Custom texts for date picker to use instead of language based texts.
+   * <pre>
+   * DatePickerTextProps {
+   *  openButtonLabel?: string;
+   *  selectedDateLabel?: string;
+   *  closeButtonText?: string;
+   *  selectButtonText?: string;
+   *  yearSelectLabel?: string;
+   *  monthSelectLabel?: string;,
+   *  nextMonthButtonLabel?: string;
+   *  prevMonthButtonLabel?: string;
+   *  monthNames?: [
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *  ];
+   *  weekDays?: [
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *  ];
+   *  weekDayAbbreviations?: [
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *  ];
+   * }
+   * </pre>
+   */
+  datePickerTexts?: DatePickerTextProps;
+  /** Language for date picker default texts. Language supports Finnish, English, and Swedish.
+   * @default 'fi'
+   */
+  language?: 'fi' | 'en' | 'sv';
```

Comment:
Since this typing is used elsewhere as well, it might be clearer to add it as custom type and use that.

Reply:
Added type for language ✔️